### ISSUE TITLE
Doctrine.ORM.EntityRepository->createQueryBuilder

### ISSUE THREAD

**Issue by joshuayoung1380** [doctrine/doctrine2#5650]
EntityRepository->createQueryBuilder does not provide to way to set the indexBy.
http://www.doctrine-project.org/api/orm/2.2/class-Doctrine.ORM.EntityRepository.html

EntityManager->createQueryBuilder allows the indexBy to be set via the from method
http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/query-builder.html#high-level-api-methods

EntityRepository->createQueryBuilder calls from for you, but does not all have an option for indexBy.

the EntityRepository->createQueryBuilder is used to in symfony's EntityType field for creating forms.
i was trying to index my arrayCollection by an id field for my form, just i preference that over a 0 indexed key.

http://symfony.com/doc/current/reference/forms/types/entity.html

Should EntityRepository's createQueryBuilder method allow for this?

**Comment by Ocramius** [comment]
@joshuayoung1380 can you rephrase your question? It is really unclear:
 1. what you are trying to do
 2. what the unexpected outcome is

**Comment by joshuayoung1380** [comment]
i may have been mistaking on how this was intended to work
